### PR TITLE
Implementation of 0/1 knapsack problem in c++

### LINKED ISSUE #1992: Implementation of 0/1 knapsack problem in C++
I want to work on this as a part of GSSoC'20. Please assign it to me if it is not implemented yet.

### PR DESCRIPTION
**Pull request by shweta0699**
Fixes #1992 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Adding implementation of 0/1 knapsack using recursion in C++

#### Languages Used:
- C++

#### Files Added:
- 01_knapsack.cpp

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by asha15** [comment]
could you modify this code to take dynamic input?

File: Knapsack/01_knapsack.cpp [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        if(weight[n-1] > capacity) {
+            return knapsack(profit, weight, capacity, n-1);
+        }
+    
+        else {
+            return max(profit[n-1] + knapsack(profit, weight, capacity-weight[n-1], n-1), knapsack(profit, weight, capacity, n-1));
+        }
+        
+    }
+
```
Comment (asha15) [inline Knapsack/01_knapsack.cpp]: remove the spaces you kept at the beginning

File: Knapsack/01_knapsack.cpp [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    int main() {
+        vector<int> profit{10, 20, 30, 44, 56, 72};
+        vector<int> weight{5, 10, 15, 25, 20, 15};
+        int capacity = 70;
+        int n = profit.size();
+        cout << knapsack(profit, weight, capacity, n);
+        return 0;
+    }
```
Comment (asha15) [inline Knapsack/01_knapsack.cpp]: remove the spaces you kept at the beginning

File: Knapsack/01_knapsack.cpp [hunk 3]
```diff
@@ -0,0 +1,47 @@
+    // Recursive implementation of 0/1 knapsack in c++
+
+    #include <iostream>
+    #include <vector>
+
```
Comment (asha15) [inline Knapsack/01_knapsack.cpp]: remove this line

File: Knapsack/01_knapsack.cpp [hunk 4]
```diff
@@ -0,0 +1,47 @@
+    // Recursive implementation of 0/1 knapsack in c++
+
+    #include <iostream>
+    #include <vector>
+    
+    using namespace std;
+
+    int knapsack(vector<int> profit, vector<int> weight, int capacity, int n) {         //function to return maximum profit
```
Comment (asha15) [inline Knapsack/01_knapsack.cpp]: take the comment into the previous line

File: Knapsack/01_knapsack.cpp [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    using namespace std;
+
+    int knapsack(vector<int> profit, vector<int> weight, int capacity, int n) {         //function to return maximum profit 
+        
+        if(capacity == 0 || n == 0 ) {
+            return 0;
+        }
+    
+        if(weight[n-1] > capacity) {
+            return knapsack(profit, weight, capacity, n-1);
```
Comment (asha15) [inline Knapsack/01_knapsack.cpp]: keep space around operator

File: Knapsack/01_knapsack.cpp [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+        cout << knapsack(profit, weight, capacity, n);
+        return 0;
+    }
+
+/*
+
+OUTPUT :
+192
+
+*/
```
Comment (asha15) [inline Knapsack/01_knapsack.cpp]: remove the extra lines

File: Knapsack/01_knapsack.cpp (hunk 3 above)
Comment (asha15) [reply Knapsack/01_knapsack.cpp]: remove the third line

Comment (asha15) [reply Knapsack/01_knapsack.cpp]: please don't keep space at the beginning

**Comment by shweta0699** [comment]
Ma'am, any updates or feedback?

**Comment by shweta0699** [comment]
Okay @asha15 ma'am.

**Comment by shweta0699** [comment]
Changes are done, ma'am. Please check and let me know if anything else is required.

**Comment by shweta0699** [comment]
@asha15 ma'am, any feedback from your side? Are there any other changes that need to be done? Please let me know ma'am.

**Comment by asha15** [comment]
please squash your commits

**Comment by shweta0699** [comment]
@asha15 ma'am, thanks for approving it. I am actually new to git and I have never really squashed commits, so can you please guide me on how to squash all the commits? That would be a great help. Thanks again.

**Comment by somya-kapoor** [comment]
For reference http://blog.steveklabnik.com/posts/2012-11-08-how-to-squash-commits-in-a-github-pull-request

**Comment by shweta0699** [comment]
@somya-kapoor ma'am, thanks for the reference. It really helped. Please review the changes again. Thanks.

**Pull request merged**